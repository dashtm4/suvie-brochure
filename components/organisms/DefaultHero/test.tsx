import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { DefaultHero } from '.'

describe('<DefaultHero />', () => {
  const title = 'Test title'

  it('should renders correct <DefaultHero /> with correct title', () => {
    const wrapper = shallow(<DefaultHero title={title} />)

    expect(wrapper.find('PageTitle').contains(title))

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <DefaultHero /> with correct subtitle', () => {
    const subtitle = 'Test subtitle'

    const wrapper = shallow(<DefaultHero title={title} subtitle={subtitle} />)

    expect(wrapper.contains(subtitle)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <DefaultHero /> with correct title and strapline', () => {
    const strapline = 'Test strapline'

    const wrapper = shallow(<DefaultHero title={title} strapline={strapline} />)

    expect(wrapper.contains(strapline)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <DefaultHero /> with background image', (done) => {
    const windowWidth = 769

    const imageSrc = 'image.jpg'
    const mobileImageSrc = 'image--small.jpg'

    const wrapper = mount(
      <DefaultHero
        title={title}
        imageSrc={imageSrc}
        mobileImageSrc={mobileImageSrc}
      />
    )

    const callback = () => {
      expect(wrapper.find('section').prop('style')).toEqual({
        backgroundImage: `url(${imageSrc})`
      })

      window.removeEventListener('resize', callback)

      expect(toJSON(wrapper)).toMatchSnapshot()

      done()
    }

    window.addEventListener('resize', callback)

    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: windowWidth
    })

    window.dispatchEvent(new Event('resize'))
  })

  it('should renders correct <DefaultHero /> with mobile background image from imageSrc prop', (done) => {
    const windowWidth = 380

    const imageSrc = 'image.jpg'

    const wrapper = mount(<DefaultHero title={title} imageSrc={imageSrc} />)

    const callback = () => {
      expect(wrapper.find('section').prop('style')).toEqual({
        backgroundImage: `url(${imageSrc})`
      })

      window.removeEventListener('resize', callback)

      expect(toJSON(wrapper)).toMatchSnapshot()

      done()
    }

    window.addEventListener('resize', callback)

    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: windowWidth
    })

    window.dispatchEvent(new Event('resize'))
  })

  it('should renders correct <DefaultHero /> with correct className when bgPosition is default', () => {
    const wrapper = shallow(<DefaultHero title={title} />)

    expect(wrapper.prop('className')).toContain('bg-center')

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <DefaultHero /> with correct className when bgPosition is top', () => {
    const bgPosition = 'top'

    const wrapper = shallow(
      <DefaultHero title={title} bgPosition={bgPosition} />
    )

    expect(wrapper.prop('className')).toContain('bg-top')

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <DefaultHero /> with correct className when bgPosition is center', () => {
    const bgPosition = 'center'

    const wrapper = shallow(
      <DefaultHero title={title} bgPosition={bgPosition} />
    )

    expect(wrapper.prop('className')).toContain('bg-center')

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <DefaultHero /> with correct className when bgPosition is bottom', () => {
    const bgPosition = 'bottom'

    const wrapper = shallow(
      <DefaultHero title={title} bgPosition={bgPosition} />
    )

    expect(wrapper.prop('className')).toContain('bg-bottom')

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
