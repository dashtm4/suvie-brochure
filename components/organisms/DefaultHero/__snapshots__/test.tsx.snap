// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultHero /> should renders correct <DefaultHero /> with background image 1`] = `
<DefaultHero
  imageSrc="image.jpg"
  mobileImageSrc="image--small.jpg"
  title="Test title"
>
  <section
    className="relative bg-secondary text-white text-center bg-center bg-cover bg-no-repeat  pt-50 pb-44 md:pt-33 md:pb-40"
    style={
      Object {
        "backgroundImage": "url(image.jpg)",
      }
    }
  >
    <Container
      className="max-w-5xl"
      role="banner"
    >
      <div
        className="container px-3 mx-auto max-w-5xl"
        role="banner"
      >
        <PageTitle>
          <h1
            className="font-serif font-normal tracking-tighter text-4xl md:text-7xl text-white"
          >
            <span
              className=""
            >
              Test title
            </span>
          </h1>
        </PageTitle>
      </div>
    </Container>
    <Wave
      colorName="white"
      position="bottom-revert"
    >
      <svg
        className="absolute left-0 w-full top-full transform -translate-y-full fill-white"
        preserveAspectRatio="xMidYMin slice"
        style={
          Object {
            "fill": "",
          }
        }
        version="1.0"
        viewBox="0 0 1200 44"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M279.243 0C174.167 0 34.974 18.59 0 27.885V44h1200V0C991.25 0 838.634 34.818 678.692 34.818S414.583 0 279.243 0z"
        />
      </svg>
    </Wave>
  </section>
</DefaultHero>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with correct className when bgPosition is bottom 1`] = `
<section
  className="relative bg-secondary text-white text-center py-20 bg-bottom"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <Container
    className="max-w-5xl"
    role="banner"
  >
    <PageTitle>
      <span
        className=""
      >
        Test title
      </span>
    </PageTitle>
  </Container>
  <Wave
    colorName="secondary"
    position="bottom"
  />
</section>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with correct className when bgPosition is center 1`] = `
<section
  className="relative bg-secondary text-white text-center py-20 bg-center"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <Container
    className="max-w-5xl"
    role="banner"
  >
    <PageTitle>
      <span
        className=""
      >
        Test title
      </span>
    </PageTitle>
  </Container>
  <Wave
    colorName="secondary"
    position="bottom"
  />
</section>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with correct className when bgPosition is default 1`] = `
<section
  className="relative bg-secondary text-white text-center py-20 bg-center"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <Container
    className="max-w-5xl"
    role="banner"
  >
    <PageTitle>
      <span
        className=""
      >
        Test title
      </span>
    </PageTitle>
  </Container>
  <Wave
    colorName="secondary"
    position="bottom"
  />
</section>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with correct className when bgPosition is top 1`] = `
<section
  className="relative bg-secondary text-white text-center py-20 bg-top"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <Container
    className="max-w-5xl"
    role="banner"
  >
    <PageTitle>
      <span
        className=""
      >
        Test title
      </span>
    </PageTitle>
  </Container>
  <Wave
    colorName="secondary"
    position="bottom"
  />
</section>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with correct subtitle 1`] = `
<section
  className="relative bg-secondary text-white text-center py-20 bg-center"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <Container
    className="max-w-5xl"
    role="banner"
  >
    <PageTitle>
      <span
        className=""
      >
        Test title
      </span>
    </PageTitle>
    <div
      className="mt-7"
    >
      Test subtitle
    </div>
  </Container>
  <Wave
    colorName="secondary"
    position="bottom"
  />
</section>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with correct title 1`] = `
<section
  className="relative bg-secondary text-white text-center py-20 bg-center"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <Container
    className="max-w-5xl"
    role="banner"
  >
    <PageTitle>
      <span
        className=""
      >
        Test title
      </span>
    </PageTitle>
  </Container>
  <Wave
    colorName="secondary"
    position="bottom"
  />
</section>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with correct title and strapline 1`] = `
<section
  className="relative bg-secondary text-white text-center py-20 bg-center"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <Container
    className="max-w-5xl"
    role="banner"
  >
    <div
      className="mb-4 text-sm font-bold tracking-wide uppercase"
    >
      Test strapline
    </div>
    <PageTitle>
      <span
        className=""
      >
        Test title
      </span>
    </PageTitle>
  </Container>
  <Wave
    colorName="secondary"
    position="bottom"
  />
</section>
`;

exports[`<DefaultHero /> should renders correct <DefaultHero /> with mobile background image from imageSrc prop 1`] = `
<DefaultHero
  imageSrc="image.jpg"
  title="Test title"
>
  <section
    className="relative bg-secondary text-white text-center bg-center bg-cover bg-no-repeat  pt-50 pb-44 md:pt-33 md:pb-40"
    style={
      Object {
        "backgroundImage": "url(image.jpg)",
      }
    }
  >
    <Container
      className="max-w-5xl"
      role="banner"
    >
      <div
        className="container px-3 mx-auto max-w-5xl"
        role="banner"
      >
        <PageTitle>
          <h1
            className="font-serif font-normal tracking-tighter text-4xl md:text-7xl text-white"
          >
            <span
              className=""
            >
              Test title
            </span>
          </h1>
        </PageTitle>
      </div>
    </Container>
    <Wave
      colorName="white"
      position="bottom-revert"
    >
      <svg
        className="absolute left-0 w-full top-full transform -translate-y-full fill-white"
        preserveAspectRatio="xMidYMin slice"
        style={
          Object {
            "fill": "",
          }
        }
        version="1.0"
        viewBox="0 0 1200 44"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M279.243 0C174.167 0 34.974 18.59 0 27.885V44h1200V0C991.25 0 838.634 34.818 678.692 34.818S414.583 0 279.243 0z"
        />
      </svg>
    </Wave>
  </section>
</DefaultHero>
`;
